locallib.php

```diff
@@ -1820,10 +1820,88 @@ function moodleoverflow_count_discussions($moodleoverflow, $course) {
 
     // Count the discussions.
     $sql = "SELECT COUNT(d.id)
-              FROM {moodleoverflow_discussions} d
-             WHERE d.moodleoverflow = ?";
+            FROM {moodleoverflow_discussions} d
+            WHERE d.moodleoverflow = ?";
     $amount = $DB->get_field_sql($sql, array($moodleoverflow->id));
 
     // Return the amount.
     return $amount;
+}
+
+
+
+
+function moodleoverflow_update_user_grade($moodleoverflow, $postuserrating, $postinguser) {
+
+    // check wheter moodleoverlfow object has the added params
+    if ($moodleoverflow->grademaxgrade > 0 and $moodleoverflow->gradescalefactor > 0) {
+
+        moodleoverflow_update_user_grade_on_db($moodleoverflow, $postuserrating, $postinguser);
+    }
+}
+
+function moodleoverflow_update_user_grade_on_db($moodleoverflow, $postuserrating, $userid) {
+    global $DB;
+
+    // calculate the posting user's updated grade
+    $grade = $postuserrating / $moodleoverflow->gradescalefactor;
+
+    if ($grade > $moodleoverflow->grademaxgrade) {
+
+        $grade = $moodleoverflow->grademaxgrade;
+    }
+
+    // save updated grade on local table
+    if ($DB->record_exists('moodleoverflow_grades', array('userid' => $userid, 'moodleoverflowid' => $moodleoverflow->id))) {
+
+        $DB->set_field('moodleoverflow_grades', 'grade', $grade, array('userid' => $userid, 'moodleoverflowid' => $moodleoverflow->id ));
+
+    } else {
+
+        $gradedataobject = new stdClass();
+        $gradedataobject->moodleoverflowid = $moodleoverflow->id;
+        $gradedataobject->userid = $userid;
+        $gradedataobject->grade = $grade;
+        $gradedataobject->late = 0;
+        $gradedataobject->completed = 0;
+        $DB->insert_record('moodleoverflow_grades', $gradedataobject, false);
+    }
+
+                // update gradebook
+    moodleoverflow_update_grades($moodleoverflow, $userid);
+}
+
+function moodleoverflow_update_all_grades_for_cm($moodleoverflowid) {
+    global $DB;
+
+    $moodleoverflow = $DB->get_record('moodleoverflow', array('id' => $moodleoverflowid));
+
+    // check wheter moodleoverlfow object has the added params
+    if ($moodleoverflow->grademaxgrade > 0 and $moodleoverflow->gradescalefactor > 0) {
+
+        // get all users id
+        $params = ['moodleoverflowid' => $moodleoverflowid];
+        $sql = 'SELECT DISTINCT p.userid
```

Comment:
Hello Justus, I want to mention that the grade was not updated for the voting user because I hadn't yet included a database query to get the mebers that had voted. So the function that updates a grade for a course module only really updates the grades of those who have commented anything in a post. I was wondering wether you had this in mind.

Reply:
Hi Andrés,
thanks for the notice, I haven't yet looked at the locallib functions in detail, I will edit it to also incorporate the raters. Also thanks for the cross-reputation fix :)